drivers/docker/coordinator.go

```diff
@@ -330,18 +336,18 @@ func (d *dockerCoordinator) removeImageImpl(id string, ctx context.Context) {
 	d.imageLock.Unlock()
 
 	for i := 0; i < 3; i++ {
-		err := d.client.RemoveImageExtended(id, docker.RemoveImageOptions{
+		_, err := d.client.ImageRemove(d.ctx, id, image.RemoveOptions{
 			Force: true, // necessary to GC images referenced by multiple tags
 		})
 		if err == nil {
 			break
 		}
 
-		if err == docker.ErrNoSuchImage {
+		if strings.Contains(err.Error(), "No such image") {
```

Comment:
They are... https://pkg.go.dev/github.com/docker/docker@v27.1.1+incompatible/errdefs
but the tricky part is that you can never be quite sure which error type the API returns, as this isn't documented (at least not in the Go SDK). So unless you sit down for a long session with a debugger, there's no way of knowing. 

I'll try to update as many checks of this sort as I can, because of course I agree it's sloppy to rely on string comparisons, but this PR has already been _a lot_ of work, and I may end up creating a follow-up ticket to make these improvements. Then again I'm in the trenches already, might as well do it.